node_modules/gh-groups/tests/util.js

```diff
@@ -20,6 +20,57 @@ var _ = require('lodash');
 var assert = require('assert');
 
 /**
+ * Assert that the groups of a user can be retrieved
+ *
+ * @param  {RestClient}         client                  The REST client to make the request with
+ * @param  {Number}             id                      The id of the user to get the memberships of
+ * @param  {Number}             [limit]                 The amount of groups that should be returned
+ * @param  {Number}             [offset]                The paging number of the results to retrieve
+ * @param  {User[]}             [expectedMemberships]   The expected memberships
+ * @param  {Function}           callback                Standard callback function
+ * @param  {OrgUnit}            callback.memberships    The members of the group
+ * @throws {AssertionError}                             Error thrown when an assertion failed
+ */
+var assertGetUserMembershipsEquals = module.exports.assertGetUserMembershipsEquals = function(client, id, limit, offset, expectedMemberships, callback) {
```

Comment:
It may not seem completely necessary now, but the following kind of method I find to usually be very helpful:

  `assertGetAllUserMemberships`

You can implement this by first getting 1 user membership, then using the count to get all of them in one page.

Once you have that, you can use it inside of `assertUpdateUserMemberships` in order to do a sanity check that:

 a) When a user is added to a group, their memberships list contains the group
 b) When a user is removed from a group, their memberships list does not contain the group